lib/blackbeard/context.rb

```diff
@@ -61,14 +61,19 @@ def user_id
     end
 
     def visitor_id
-      @visitor_id ||= (controller.request.cookies['bbd'] || generate_visitor_id).to_i
+      @visitor_id ||=
+        if controller
+          controller.request.cookies['bbd'].to_i
```

Comment:
is this functionality change intentional? If the cookie is empty the visitor will be 0 which doesn't seem right.